builtin/providers/ns1/permissions.go

```diff
@@ -0,0 +1,309 @@
+package ns1
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"gopkg.in/ns1/ns1-go.v2/rest/model/account"
+)
+
+func permissionsSchema() *schema.Schema {
+	return &schema.Schema{
```

Comment:
Im not sure if im using the `helper` package correctly with the `permissions` schema here. This may a regression i introduced from the previous implementations of the ns1 provider.  

Im thinking that @bobtfish @sarguru (original authors) have much more wisdom than me, because they had orignally implemented permissions as such:

```
permissions = {
  dns_view_zones = true
  data_manage_datafeeds = true
}
```

Which works without error. 

But then i tried to mimic the ns1 api in making the individual permissions structures nested, but in doing so, caused plans to never be empty:


```
permissions = {
  dns = {
    view_zones = true
  }
  data = {
  	manage_datafeeds = true
  }
}
```


```
=== RUN   TestAccNS1Team_Basic
--- FAIL: TestAccNS1Team_Basic (7.42s)
        testing.go:265: Step 0 error: Check failed: Check 5/5 error: ns1_team.foobar: Attribute 'permissions.dns.#' expected "1", got ""
=== RUN   TestAccNS1Team_Updated
--- FAIL: TestAccNS1Team_Updated (9.47s)
        testing.go:265: Step 0 error: After applying this step, the plan was not empty:

                DIFF:

                UPDATE: ns1_team.foobar
                  permissions.0.account.#:    "1" => "0"
                  permissions.0.data.#:       "1" => "0"
                  permissions.0.monitoring.#: "1" => "0"
```

I can easily move the `permissions` structure back, but just wanted to know if it was possible to do this the 'nested' way. Or if nested structures should be avoided due to dag construction/diffs/all of the complicated cool stuff that the terraform library does for us.